### DISCUSSION TITLE
DAI 2024-2025 - Practical work 3 - David Schildböck, Kénan Augsburger and Arno Tribolet

### DISCUSSION THREAD

**Discussion by nyxhawk** [heig-vd-dai-course/heig-vd-dai-course#549] [Show and tell]
# CTFman, a CTF manager (with [BDR](https://github.com/BDR-A-24F-HEIG-VD/bdr-projet-schildbock_augsburger_tribolet) - created with GitHub Classroom; not sure if others have access)

## Authors

- [David Schldböck](https://github.com/shadowkudo/)
- [Kénan Augsburger](https://github.com/Mondotosz)
- [Arno Tribolet](https://github.com/arnoheigvd/)

### BDR subject (Our specifications document and the UML diagram (see below) will be uploaded to our DAI repository for reference.)
In the BDR course, we decided to create a database to manage CTFs (Capture The Flag) - a competition where participants solve cybersecurity challenges to gain points and compete against others. The database will help us efficiently organize and track participants, challenges, scores, and team progress, providing a structured way to analyze and improve the competition's outcomes.

### UML - Version as of December 17, 2024
![image](https://github.com/user-attachments/assets/6f6b3ccc-917d-4595-b8cb-40d3a4806776)

### DAI scope
For this project, we will implement the **backend** to manage CTFman, including a comprehensive **API**. If time allows, we will also develop a **frontend** using [SvelteKit](https://svelte.dev/) in static mode.

Endpoints to implement initially:
- Login: To access CTF or challenge content based on user role.
- Challengers: To manage or view challenger information.
- Teams: To manage or view team details.
- CTF: To manage or view CTF competitions.
- Challenge: To manage or view challenges within the CTF.

If time permits, additional endpoints to complete:
- Comment: To manage or view comments on challenges or CTFs.
- Tool: To manage or view tools associated with challenges.
- Vulnerability: To manage or view vulnerabilities related to challenges.

## Setup
GitHub repo &rarr; [here](https://github.com/shadowkudo/ctfman)
Last commit &rarr; [0e17d1d](https://github.com/shadowkudo/ctfman/commit/0e17d1db3717b9fda95f717ffc2b72fdfcbf70f8)
Tag &rarr; [V1.0.0](https://github.com/shadowkudo/ctfman/releases/tag/V1.0.0)

**Comment by ludelafo** [comment]
Good for us, you can even only manage two (2) resources in the context of this course if you want.
